evennia/contrib/rpg/rpsystem/rpsystem.py

```diff
@@ -1635,21 +1487,23 @@ def get_display_name(self, looker, **kwargs):
                 said object.
 
         Notes:
-            The RPCharacter version of this method colors its display to make
+            The RPCharacter version adds additional processing to sdescs to make
             characters stand out from other objects.
 
         """
-        idstr = "(#%s)" % self.id if self.access(looker, access_type="control") else ""
+        idstr = "(#%s)" % self.id if self.access(looker, access_type="control") and not kwargs.get("no_id",False) else ""
+        ref = kwargs.get("ref","~")
+    
         if looker == self:
-            sdesc = self.key
+            sdesc = self.process_recog(self.key,self)
         else:
             try:
-                recog = looker.recog.get(self)
+                sdesc = looker.get_sdesc(self, process=True, ref=ref)
             except AttributeError:
-                recog = None
-            sdesc = recog or (hasattr(self, "sdesc") and self.sdesc.get()) or self.key
+                sdesc = self.sdesc.get()
         pose = " %s" % (self.db.pose or "is here.") if kwargs.get("pose", False) else ""
-        return "|c%s|n%s%s" % (sdesc, idstr, pose)
+        return "%s%s%s" % (sdesc, idstr, pose)
```

Comment:
Why not get rid of the `%s` formatting when you are at it? Also, if the intent of this refactor is (in part) to make this easier to override and customize, maybe it would be an idea to have these arguments pass through a helper method you can override instead?